src/mlpack/methods/reinforcement_learning/environment/mountain_car.hpp

```diff
@@ -0,0 +1,172 @@
+/**
+ * @file mountain_car.hpp
+ * @author Shangtong Zhang
+ *
+ * This file is an implementation of Mountain Car task
+ * https://gym.openai.com/envs/MountainCar-v0
+ *
+ * TODO: refactor to OpenAI interface
```

Comment:
Oh maybe this comment is kind of misleading, I didn't mean to have same API as OpenAI gym. I just want to provide an alternative implementation, i.e. the logic of `Sample` can also be done via OpenAI gym if user specifies this.